backend/src/ee/services/group/group-fns.ts

```diff
@@ -285,6 +287,67 @@ export const addUsersToGroupByUserIds = async ({
   });
 };
 
+/**
+ * Add identities with identity ids [identityIds] to group [group].
+ * @param {group} group - group to add identity(s) to
+ * @param {string[]} identityIds - id(s) of organization scoped identity(s) to add to group
+ */
+export const addIdentitiesToGroup = async ({
+  group,
+  identityIds,
+  identityDAL,
+  identityOrgMembershipDAL,
+  identityGroupMembershipDAL
+}: TAddIdentitiesToGroup) => {
+  const processAddition = async (tx: Knex) => {
+    const identityIdsSet = new Set(identityIds);
+    const identityIdsArray = Array.from(identityIdsSet);
+
+    const foundIdentities = await identityOrgMembershipDAL.findByIds(identityIdsArray, group.orgId, tx);
+
+    const existingIdentityOrgMembershipsIdentityIdsSet = new Set(foundIdentities.map((u) => u.id));
+
+    identityIdsArray.forEach((identityId) => {
+      if (!existingIdentityOrgMembershipsIdentityIdsSet.has(identityId)) {
+        throw new ForbiddenRequestError({
+          message: `Identity with id ${identityId} is not part of the organization`
+        });
+      }
+    });
+
+    // check if identity group membership already exists
+    const existingIdentityGroupMemberships = await identityGroupMembershipDAL.find(
+      {
+        groupId: group.id,
+        $in: {
+          identityId: identityIdsArray
+        }
+      },
+      { tx }
+    );
+
+    if (existingIdentityGroupMemberships.length) {
+      throw new BadRequestError({
+        message: `${identityIdsArray.length > 1 ? `Identities are` : `Identity is`} already part of the group ${group.slug}`
+      });
+    }
+
+    await identityGroupMembershipDAL.insertMany(
+      foundIdentities.map((identity) => ({
+        identityId: identity.id,
+        groupId: group.id
+      })),
+      tx
+    );
+
+    return foundIdentities;
+  };
+
+  return identityDAL.transaction(async (tx) => {
+    return processAddition(tx);
```

Comment:
If we are doing this only once - why do we need this  as function?

Reply:
should I remove the explicit function and write everything within the transaction?